lib/gen_stage/metrics.ex

```diff
@@ -0,0 +1,15 @@
+defmodule GenStage.Metrics do
+  @moduledoc false
+
+  use Instruments.CustomFunctions, prefix: "genstage"
+
+  @doc """
+  Returns the registered name of the current process or nil.
+  """
+  def metrics_self_name() do
```

Comment:
whats the diffo between this & `Utils.self_name/0`?

Reply:
This one is used to get a name to use in the metric tags. 
`Utils.self_name` returns a PID when there's no registered name, which would blow up the metric's cardinality. This one just returns `nil` in that case so that we don't add a tag.

Reply:
got it!  that makes good sense.

Reply:
I just changed this to use "no_registered_name" instead of `nil`, so that the tags are more consistent. It should make setting up datadog dashboards a bit simpler